sympy/matrices/matrices.py

```diff
@@ -1549,11 +1549,12 @@ def jordan_cell_power(jc, n):
             l = jc[0, 0]
             if (num < 0) == True and l == 0:
                 raise ValueError("Matrix det == 0; not invertible")
-            if (num < 2) == True:
-                b = self.jordan_cells()[1]
-                for i in range(len(b)):
-                    if (b[i]).rows > 1 and b[i][0, 0] == 0:
-                        raise ValueError("Cannot be evaluated")
+            if not isinstance(num, (int, Integer, Expr)):
+                if not num - int(num) == 0:
```

Comment:
If `num` were a SymPy expression, it is would be possible to ask `if not num.is_integer`.
Since this is an internal method, the exponent could be sympified at entrance to `__pow__`.